cognite_toolkit/_cdf_tk/loaders/_resource_loaders/auth_loaders.py

```diff
@@ -317,18 +321,41 @@ def _upsert(self, items: Sequence[GroupWrite]) -> GroupList:
         # We MUST retrieve all the old groups BEFORE we add the new, if not the new will be deleted
         old_groups = self.client.iam.groups.list(all=True)
         old_group_by_names = {g.name: g for g in old_groups.as_write()}
-        changed = []
+        to_create = []
         for item in items:
             if (old := old_group_by_names.get(item.name)) and old == item:
-                # Ship unchanged groups
+                # Skip unchanged groups
                 continue
-            changed.append(item)
-        if len(changed) == 0:
+            to_create.append(item)
+        if len(to_create) == 0:
             return GroupList([])
-        created = self.client.iam.groups.create(changed)
+        created = self.client.iam.groups.create(to_create)
         created_names = {g.name for g in created}
         to_delete = [g.id for g in old_groups if g.name in created_names and g.id]
-        self.client.iam.groups.delete(to_delete)
+        failed_deletes = []
+        error_str = ""
+        try:
+            self.client.iam.groups.delete(to_delete)
```

Comment:
Not sure how it is possible to get a `CogniteNotFoundError` here, as the `to_delete` is obtained from the `self.client.iam.groups.list(all=True)`. However, looking at the traceback from the user that is exactly what has happened. Thus implementing this robustification.